src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -112,6 +114,33 @@ public string ConfigFilePath
     /// <exception cref="DataApiBuilderException">Thrown when the loader is unable to load an instance of the config from its known location.</exception>
     public RuntimeConfig GetConfig()
     {
+        // Only used in hot reload to validate the configuration file
+        if (_configLoader.IsNewConfigDetected && !_configLoader.IsNewConfigValidated)
+        {
+            IFileSystem fileSystem = new FileSystem();
+            ILoggerFactory loggerFactory = new LoggerFactory();
+            ILogger<RuntimeConfigValidator> logger = loggerFactory.CreateLogger<RuntimeConfigValidator>();
+            RuntimeConfigValidator runtimeConfigValidator = new(this, fileSystem, logger, true);
+
+            _configLoader.IsNewConfigDetected = false;
+            _configLoader.IsNewConfigValidated = runtimeConfigValidator.TryValidateConfig(ConfigFilePath, loggerFactory).Result;
+
+            // Saves the lastValidRuntimeConfig as the new RuntimeConfig if it is validated for hot reload
+            if (_configLoader.IsNewConfigValidated)
+            {
+                _configLoader.CreateNewLkgConfig();
+            }
+            else
+            {
+                _configLoader.RestoreLkgConfig();
+
```

Comment:
Put the `_configLoader.IsNewConfigDetected = false` inside of `_configLoader.DoesConfigNeedValidation()` because it needs to go before `TryValidateConfig` as if it were to go after it would cause an infinite loop.